components/script/dom/document.rs

```diff
@@ -4598,6 +4602,24 @@ impl Document {
     pub(crate) fn is_initial_about_blank(&self) -> bool {
         self.is_initial_about_blank.get()
     }
+
+    pub(crate) fn highlight_dom_node(&self, node: Option<&Node>) {
+        let Some(root_element) = self.GetDocumentElement() else {
+            return;
+        };
+
+        self.highlighted_dom_node.set(node);
+
+        // TODO: Node highlights don't affect layout.
+        // Invent a less destructive way to show them
+        root_element
+            .upcast::<Node>()
+            .dirty(NodeDamage::OtherNodeDamage);
```

Comment:
How about `document.set_needs_paint(true)`?

Reply:
I didn't know that method existed, thanks.